### DISCUSSION TITLE
Want to know semantics of blockages

### DISCUSSION THREAD

**Discussion by doo3yoon** [The-OpenROAD-Project/OpenROAD#5249] [Q&A]
Hi

I applied routing blockages before global routing stage using command lines below for the whole design area,

set_global_routing_region_adjustment [list 0.1 0.1 237.4 140] -layer 2 -adjustment 0.2
set_global_routing_region_adjustment [list 0.1 0.1 237.4 140] -layer 3 -adjustment 0.2
set_global_routing_region_adjustment [list 0.1 0.1 237.4 140] -layer 4 -adjustment 0.2
set_global_routing_region_adjustment [list 0.1 0.1 237.4 140] -layer 5 -adjustment 0.2

and compared its routing resource reduction with the one without routing blockage (refer to the table below).

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/102841960/69b52af8-806c-4f18-bb1e-4059ade2d953)

But, I couldn't find a clear connection between blockage value and resource reduction.
For example, when I reduce 20% for metal2, its derated resource was from 0.89% to 29.75%.

Could you explain about how the routing blockage affects routing resource reduction?

Thanks and looking forward to hearing from you,

Dooseok

**Comment by maliberty** [comment]
@eder-matheus please comment

**Comment by eder-matheus** [comment]
@doo3yoon The resource reduction you see in the run without applying `set_global_routing_region_adjustment` comes from the design obstructions, like PWR/GND nets, instance obstructions, macros, etc.

The percentage shown in the report does not match your reduction because it operates over integer values, and we always floor the result. So the global router will always be pessimistic with your reduction value, which can lead to a higher reduction percentage.

You can see that the higher layers have more impact because their pitch values are already high, meaning that the original resources will be smaller when compared to the lower layers. When we floor the resources after applying the reduction for these upper layers, we could end with a more significant reduction than the other layers.
